salt/modules/mount.py

```diff
@@ -1422,3 +1569,287 @@ def delete_mount_cache(real_name):
                 if not cache_write:
                     raise CommandExecutionError('Unable to write mount cache.')
     return True
+
+
+def _filesystems(config='/etc/filesystems', leading_key=True):
+    '''
+    .. versionadded:: 2018.3.3
+
+    List the contents of the filesystems
+
+    config
+        File containing filesystem infomation
+
+    leading_key
+        True    return dictionary keyed by 'name' value and dictionary with other keys, values
+                { '/dir' : { 'dev': '/dev/hd8', .... }}
+
+        False   return dictionary  keyed by 'name' value and dictionary with all keys, values
+                { '/dir' : { 'name' : '/dir', 'dev': '/dev/hd8', .... }}
+    '''
+    ret = OrderedDict()
+    lines = []
+    parsing_block = False
+    if not os.path.isfile(config) or 'AIX' not in __grains__['kernel']:
+        return ret
+
+    # read in block of filesystems, block starts with '/' till empty line
+    with salt.utils.files.fopen(config) as ifile:
+        for line in ifile:
+            line = salt.utils.stringutils.to_unicode(line)
+
+            # skip till first entry
+            if not line.startswith('/') and not parsing_block:
+                continue
+
+            if line.startswith('/'):
+                parsing_block = True
+                lines.append(line)
+            elif not line.split():
+                parsing_block = False
+                try:
+                    entry = _filesystems_entry.dict_from_lines(
+                        lines,
+                        _filesystems_entry.compatibility_keys)
+                    lines = []
+                    if 'opts' in entry:
+                        entry['opts'] = entry['opts'].split(',')
+                    while entry['name'] in ret:
+                        entry['name'] += '_'
+
+                    if leading_key:
+                        ret[entry.pop('name')] = entry
+                    else:
+                        ret[entry['name']] = entry
+
+                except _filesystems_entry.ParseError:
+                    pass
+            else:
+                lines.append(line)
+
+    return ret
+
+
+def filesystems(config='/etc/filesystems'):
+    '''
+    .. versionadded:: 2018.3.3
+
+    List the contents of the filesystems
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mount.filesystems
+    '''
+    ret = {}
+    if 'AIX' not in __grains__['kernel']:
+        return ret
```

Comment:
We can't just silently do this. It's going to appear completely broken to nearly all users. It either needs to be implemented for the operating systems supported by this module or not implemented at all.

Reply:
I think that this is meant as an AIX complement to the `fstab` function used on other platforms. It should be explicitly documented as such to distinguish itself.

I don't understand the reasoning for putting all the work in a separate, private function, though.

Reply:
@terminalmage reason for separate functions for filesystems, set_filesystems, rm_filesystems is following established pattern, for example: automaster, set_automaster, rm_automaster and similar for vfstab, even if it basically call fstab equivalent with a different config file.  In this case more so, since /etc/filesystems (multi-line entries), is not even close to /etc/fstab in form (single line entries).

@cachedout the quick exit is similar to that already implemented, for example def automaster checks for a file (/etc/auto_salt) which will probably not exist on Linux and hence quickly exit. I could do the same check for file, however I feel the OS check is more inherently accurate as to functionality exists or not.  That said, I can change it to a file test similar to automaster.
Let me know your preference.

Reply:
No that's not what I was talking about. I was talking about the fact that all the actual work was in a `_filesystems` function.

Reply:
@terminalmage the use of _filesystems function is only to read in the contents /etc/filesystems, in order, and then allow manipulation as desired, here to write the new state to /etc/filesystems in the correct order.  Order is significant on AIX and should not be changed, otherwise operations are affected.
Hence the use of _filesystems is to obtain the current /etc/filesystems into an instance where it can be manipulated.  The function is used in plain output, setting and removing. A common internal function used, perhaps you could explain further your objection to its use.

Partial contents of /etc/filesystems for example:

/opt:
        dev             = /dev/hd10opt
        vfs             = jfs2
        log             = /dev/hd8
        mount           = true
        check           = true
        vol             = /opt
        free            = false
        quota           = no

/var/adm/ras/livedump:
        dev             = /dev/livedump
        vfs             = jfs2
        log             = /dev/hd8
        mount           = true
        account         = false
        quota           = no

Reply:
I did not catch the other calls to the function in my earlier review.